apolloschurchapp/src/tabs/Tab.js

```diff
@@ -0,0 +1,87 @@
+import React, { useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { useApolloClient, gql, useQuery } from '@apollo/client';
+
+import { BackgroundView, NavigationService } from '@apollosproject/ui-kit';
+import {
+  FeaturesFeedConnected,
+  FEATURE_FEED_ACTION_MAP,
+  RockAuthedWebBrowser,
+} from '@apollosproject/ui-connected';
+
+import { checkOnboardingStatusAndNavigate } from '@apollosproject/ui-onboarding';
+import { createNativeStackNavigator } from 'react-native-screens/native-stack';
+
+import { ONBOARDING_VERSION } from '../ui/Onboarding';
+
+function handleOnPress({ action, ...props }) {
+  if (FEATURE_FEED_ACTION_MAP[action]) {
+    FEATURE_FEED_ACTION_MAP[action]({ action, ...props });
+  }
+}
+
+export const createTab = ({ tabName, screenOptions, ...props }) => {
+  const TabStack = createNativeStackNavigator();
+  const TabComponent = () => <Tab tab={tabName.toUpperCase()} />;
+  const TabNav = () => (
+    <TabStack.Navigator
+      screenOptions={{
+        headerHideShadow: true,
+        headerLargeTitle: true,
+        ...screenOptions,
+      }}
+    >
+      <TabStack.Screen name={tabName} component={TabComponent} {...props} />
```

Comment:
How might I pass additional routes?

Reply:
And what does spreading props here do differently then spreading props in the component def above?

Reply:
Removed spreading of props. If a user wants to pass additional props they shouldn't use this function.